lib/ansible/modules/monitoring/grafana_dashboard.py

```diff
@@ -251,10 +270,20 @@ def grafana_create_dashboard(module, data):
         else:
             uid = None
 
+    result = {}
+
+    # test if the folder exists
+    folder_exists = grafana_folder_exists(module, data['grafana_url'], data['folder_id'], headers)
+    if folder_exists is False:
+        result['msg'] = "Dashboard folder %d does not exist." % data['folder_id']
+        result['uid'] = uid
+        result['changed'] = False
+        return result
```

Comment:
Unsure if this is the correct style. E.g. if this should be an exception, or a silent _skip_.

Reply:
Maybe the solution ils to give a folder name instead of an id. And create it if not exists ?
Unless, you can add "the dashboard import ils skipped" un the output message

Reply:
@seuf I can get folders by `:id` (unique to the Grafana install), or `:uid` (supposedly globally unique). If I take a "folder name", I'd have to parse the output to find it. Not entirely sure if that makes everything too slow. I think names are not "unique", there are more titles according to the API:

https://grafana.com/docs/http_api/folder/

Can you clarify what you mean with the "skipped" part? Should I change the message, `Dashboard folder %d does not exist` to, `Dashboard import was skipped`?

Reply:
Providing a folder name would mean that the module support providing path like "first_dir/sub_dir" and would find the id of "sub_dir". As stated by @till, a folder name can be repeated in several sub directories.